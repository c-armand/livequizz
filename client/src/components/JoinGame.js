import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { joinGame } from '../actions/playersActions'

class JoinGame extends Component {
  constructor(props) {
    super(props)
    this.state = {
      username: ''
    }
    this.onChange = this.onChange.bind(this)
    this.onSubmit = this.onSubmit.bind(this)
  }

  onChange(e) {
    this.setState({ username: e.target.value})
  }

  onSubmit(e) {
    e.preventDefault()
    this.props.dispatch({ username: this.state.username })
    this.setState({ username: '' })
  }

  render() {
    return (
      <div className="JoinGame p-3">
        <form onSubmit={this.onSubmit}>
          <div className="form-group">
            <input
              type="text"
              name="username"
              placeholder="Votre nom"
              className="form-control"
              value={this.state.username}
              onChange={this.onChange}
            />
          </div>
          <div className="form-group mb-0">
            <button className="btn btn-primary btn-block shadow-sm">Jouer !</button>
          </div>
        </form>
      </div>
    )
  }
}

JoinGame.propTypes = {
  dispatch: PropTypes.func.isRequired
}

const mapDispatchToProps = dispatch => ({
  dispatch: (player) => {
    dispatch(joinGame(player))
  }
})

export default connect(() => ({}), mapDispatchToProps)(JoinGame)
