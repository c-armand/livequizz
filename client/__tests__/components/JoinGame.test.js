import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import JoinGame from '../../src/components/JoinGame'

describe('<JoinGame />', () => {
  it('renders the component', () => {
    const component = shallow(<JoinGame />)
    // const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot()
  })
})