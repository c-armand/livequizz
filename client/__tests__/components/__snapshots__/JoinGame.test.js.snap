// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JoinGame /> renders the component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
